House-of-glass/src/Contexts/Product/ProductCompose.jsx

```diff
@@ -0,0 +1,59 @@
+const sortby = ({ sortBy }, array) => {
+  switch (sortBy) {
+    case "HIGHTOLOW":
+      return sortBy === "HIGHTOLOW"
+        ? [...array].sort((a, b) => Number(b.price) - Number(a.price))
+        : [...array];
+    case "LOWTOHIGH":
+      return sortBy === "LOWTOHIGH"
+        ? [...array].sort((a, b) => Number(a.price) - Number(b.price))
+        : [...array];
+    default:
+      return [...array];
+  }
+};
+
+const fastdelivery = ({ fastDelivery }, array) => {
+  let filteredList = fastDelivery
+    ? array.filter((item) => item.fastDelivery)
+    : array;
+  return filteredList;
+};
+
+const outofstock = ({ outOfStock }, array) => {
+  let filteredList = outOfStock
```

Comment:
Filter is a pure function. Could have directly returned it

Reply:
Yes, thanks for pointing this out. Have corrected it in latest commit.